frontend/src/app/application-header/application-header.component.ts

```diff
@@ -0,0 +1,63 @@
+import { AfterViewInit, ChangeDetectionStrategy, Component, OnDestroy, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-application-header',
+  templateUrl: './application-header.component.html',
+  styleUrls: ['./application-header.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ApplicationHeaderComponent implements OnInit, AfterViewInit, OnDestroy {
+  lastScrollPosition: number = 0;
+  PUZZLE_TOP_BAR_HEIGHT: number = 48;
+  okrBanner: HTMLElement | null = null;
+  eventListener: EventListener | null = null;
+
+  resizeObserver: ResizeObserver = new ResizeObserver((entries: ResizeObserverEntry[]) => {
+    this.removeScrollEventListener();
+    this.setScrollEventListener(entries[0].contentRect.height);
+  });
+
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  ngAfterViewInit(): void {
+    //Define puzzleBanner reference and add resizeObserver
+    this.okrBanner = document.getElementById('okrBanner')!;
+    this.resizeObserver.observe(this.okrBanner);
+  }
+
+  ngOnDestroy(): void {
+    this.removeScrollEventListener();
```

Comment:
add `this.resizeObserver.disconnect();` for remove all listeners

Reply:
Disconnected